media/system/js/core.js

```diff
@@ -1,1131 +1,933 @@
-		var checkWC = function (wc) {
-			if (wc.hasOwnProperty('fullPolyfill') && wc['fullPolyfill'] === true) {
-				return true;
-			}
-			return false;
-		};
-
-		/* Load web components async */
-		var loadWC = function (wc) {
-			var el, p, es5;
-			for (p in wc) {
-				if (wc.hasOwnProperty(p) && p !== 'fullPolyfill') {
-					if (wc[p].match(/\.js/g)) {
-						el = document.createElement('script');
-						if (!checkES6()) {
-							// Browser is not ES6!
-							if (wc[p].match(/\.min\.js/g)) {
-								es5 = wc[p].replace(/\.min\.js/g, '-es5.min.js')
-							} else if (wc[p].match(/\.js/g)) {
-								es5 = wc[p].replace(/\.js/g, '-es5.js')
-							}
-							el.src = es5;
-						} else {
-							el.src = wc[p];
-						}
-					}
-					if (el) {
-						document.head.appendChild(el);
-					}
-				}
-			}
-		};
-
-		if (checkWC(wc)) {
-			if (!('import' in document.createElement('link'))) {
-				polyfills.push('hi');
-			}
-			if (!('attachShadow' in Element.prototype && 'getRootNode' in Element.prototype) || (window.ShadyDOM && window.ShadyDOM.force)) {
-				polyfills.push('sd');
-			}
-			if (!window.customElements || window.customElements.forcePolyfill) {
-				polyfills.push('ce');
-			}
-			if (!('content' in document.createElement('template')) || !window.Promise || !Array.from || !(document.createDocumentFragment().cloneNode() instanceof DocumentFragment)) {
-				polyfills = ['lite'];
-			}
-		} else {
-			if (!window.customElements || window.customElements.forcePolyfill) {
-				polyfills.push('ce');
-			}
-		}
-
-		if (polyfills.length) {
-			var name = "core.min.js", script = document.querySelector('script[src*="' + name + '"]');
-
-			if (!script) {
-				name = "core.js";
-				script = document.querySelector('script[src*="' + name + '"]')
-			}
-
-			if (!script) {
-				throw new Error('core(.min).js is not registered correctly!')
-			}
-
-			var newScript = document.createElement('script'),
-			    replacement = 'media/system/js/polyfills/webcomponents/webcomponents-' + polyfills.join('-') + '.min.js',
-			    mediaVersion = script.src.match(/\?.*/)[0],
-			    base = Joomla.getOptions('system.paths');
-
-			if (!base) {
-				throw new Error('core(.min).js is not registered correctly!')
-			}
-
-			newScript.src = base.rootFull + replacement + (mediaVersion ? mediaVersion : '');
-			document.head.appendChild(newScript);
-
-			document.addEventListener('WebComponentsReady', function () {
-				loadWC(wc);
-			});
-		} else {
-			var fire = function () {
-				requestAnimationFrame(function () {
-					document.dispatchEvent(new CustomEvent('WebComponentsReady', { bubbles: true }));
-					loadWC(wc);
-				});
-			};
-
-			if (document.readyState !== 'loading') {
-				fire();
-			} else {
-				document.addEventListener('readystatechange', function wait() {
-					fire();
-					document.removeEventListener('readystatechange', wait);
-				});
-			}
-		}
-	};
-}( Joomla, document ));
+    /* Load web components async */
+    const loadWC = (wc) => {
```

Comment:
Surely this can't be an arrow function if you're checking or ES6 compat below...

Reply:
yup here it makes no sense, but it makes sense in the transpiled code